ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/FieldInfo.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client;
+
+/**
+ * Represents a field/column of a ksqlDB stream/table.
+ */
+public interface FieldInfo {
+
+  /**
+   * @return name of this field
+   */
+  String name();
+
+  /**
+   * @return type of this field
+   */
+  ColumnType type();
```

Comment:
The server endpoint actually returns nested field type information in a structured way, so `describeSource` could easily return nested field types, i.e., show the fields and types within arrays, maps, and structs, rather than simply showing "ARRAY", "MAP", and "STRUCT". However, by sharing this `ColumnType` interface that's also used by in the `streamQuery()` and `executeQuery()` methods, we've coupled the two together -- we either have to:
1. Expose nested type information for `describeSource()` and also `streamQuery()`/`executeQuery()`, the latter of which is not as easy to do since the server endpoint used there returns type information as a string rather than in structured format.
1. Do not expose nested type information for either `describeSource()` or `streamQuery()`/`executeQuery()`, as is currently the case in this PR
1. Do not share the `ColumnType` interface between these different client methods.

I don't have a sense of how important it is for users to receive nested type information from `describeSource()`. Unless we think it's important, my vote is option (2) for now.